services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy

```diff
@@ -45,4 +51,100 @@ class SwitchesSpec extends BaseSpecification {
         e.statusCode == HttpStatus.NOT_FOUND
         e.responseBodyAsString.to(MessageError).errorMessage == "Switch $NON_EXISTENT_SWITCH_ID not found."
     }
+
+    def "Systems allows to get a flow that goes through a switch"() {
+        given: "Two active not neighboring switches with two diverse paths at least"
+        def switchPair = topologyHelper.getAllNotNeighboringSwitchPairs().find {
+            it.paths.unique(false) { a, b -> a.intersect(b) == [] ? 1 : 0 }.size() >= 3
+        } ?: assumeTrue("No suiting switches found", false)
+
+        and: "A protected flow"
+        def protectedFlow = flowHelper.randomFlow(switchPair)
+        protectedFlow.allocateProtectedPath = true
+        flowHelper.addFlow(protectedFlow)
+
+        and: "A single switch flow"
+        def simpleFlow = flowHelper.singleSwitchFlow(switchPair.src)
+        flowHelper.addFlow(simpleFlow)
```

Comment:
You should ensure this port is not the same as in `protectedFlow`, otherwise your 'port' searches may return more results than you expect

Reply:
This is not enough. Current implementation only guarantees 'no conflict' situation, but two flows may still be located on the same port, but with different vlans.

Reply:
now it is totally fixed)